tools/topic_tools/src/throttle.cpp

```diff
@@ -126,6 +126,11 @@ void in_cb(const ros::MessageEvent<ShapeShifter>& msg_event)
         now.fromSec(ros::WallTime::now().toSec());
       else
         now = ros::Time::now();
+      if (g_last_time > now)
+      {
+        ROS_WARN("Detected jump back in time.");
```

Comment:
This should state what it's doing, not just that it detected. Something like "Detected jump back in time, resetting throttle period to now for ***."

Reply:
@mikepurvis @tfoote Thank you very much for your review!

@tfoote I updated the message. Please check. :+1: